### PR TITLE
Weapons batch 1

### PR DESCRIPTION
**Pull request by s-schmitt**
closes issue #36

### CODE REVIEW

**Comment by s-schmitt** [comment]
Build: http://mocsarcade.github.io/starjunk/weapons.a/

**Comment by s-schmitt** [comment]
https://mocsarcade.github.io/starjunk/HDRemaster.a

checkout weapon "0"

File: source/PowerUp.js [hunk 1]
```diff
@@ -97,8 +97,286 @@ export class FiveShoota extends PowerUp {
...
+
+}
+
+export class RapidSprayShot extends PowerUp {
+
+      constructor() {
+          super()
+          this.vector = new Victor(1,0)
+          this.vectorN = new Victor(1,0.25)
+          this.vectorP = new Victor(1,-0.25)
```
Comment (Phyoog) [inline source/PowerUp.js]: What's the naming convention here? This is obviously the middle, top, and bottom projectiles, but what are the `N` and `P` denoting?

File: source/PowerUp.js [hunk 2]
```diff
@@ -97,8 +97,286 @@ export class FiveShoota extends PowerUp {
...
+          this.bulletSpeed = 1
+          this.projectileType = "bullet"
+          this.rapidFire = true
+      }
+
+        fire(curShip) {
+
+            this.vector.normalize()
+            this.vectorN.normalize()
+            this.vectorP.normalize()
```
Comment (Phyoog) [inline source/PowerUp.js]: Can we not normalize these once in the constructor and then not call it again anymore? Seems a waste to keep normalizing the vectors every single time we fire.

File: source/PowerUp.js [hunk 3]
```diff
@@ -108,7 +386,7 @@ export class RapidFire extends PowerUp {
           game.addChild(new Projectile(
               curShip.position.x + curShip.width,
               curShip.position.y + curShip.height/2,
-              curShip, this.fireRate, this.bulletSpeed, this.projectileType))
+              this.vector, curShip, this.bulletSpeed, this.projectileType))
       }
```
Comment (Phyoog) [inline source/PowerUp.js:396]: The BFG is supposed to have kickback equal to the ship's normal forward velocity, so you'll need to add something like `curShip.move(-curShip.speed * delta, "x")`

**Comment by s-schmitt** [comment]
Ok everything should be fixed now. The BFG kick back uses a magic number for now because it still needs to be balanced as far as fire rate.

**Comment by ehgoodenough** [comment]
@HolySchmitt; wooooow! HDremaster Weapon 0 is trippy. :D

**Comment by s-schmitt** [comment]
https://mocsarcade.github.io/starjunk/weaponfixes.a

Here is with the BFG kick back

**Comment by s-schmitt** [comment]
Yeah Dustin and I wanted to bring up the point that some things in the game could benefit from a larger resolution. Not as big as I have it there but a slightly larger screen could mean more room to dodge and to fill with enemies.

**Comment by ehgoodenough** [comment]
I think a larger resolution will also hide the issue we've been having with pixel jumping.

**Comment by s-schmitt** [comment]
May a quick change to rapid fire and the BFG to slow down their fire rates.

**Comment by ehgoodenough** [comment]
@HolySchmitt Can we get a playable build?

**Comment by s-schmitt** [comment]
https://mocsarcade.github.io/starjunk/update.a

**Comment by ehgoodenough** [comment]
We need to get better at naming our playable builds :| (no need to add `.a`, right?)

**Comment by ehgoodenough** [comment]
So what's the difference between 1 and 4 besides speed? And what is 0?

**Comment by s-schmitt** [comment]
1) Regular Pea Shooter
4) Rapid Fire (can hold space bar down)
0) Rapid Fire Spray Shot

**Comment by Phyoog** [comment]
Kickback on the BFG feels good. I like the shudder it gives. With a really awesome sound effect, that gun is going to feel awesome.

I'd like to merge this, but I'm worried about carrying forward the technical debt for adding delta to the BFG's kickback mechanic. I'll bring it up in the face-to-face this morning and we'll take care of it.

**Comment by Phyoog** [comment]
Looks good. Merging!

[selfie-0](https://github.com/thieman/github-selfies/)

**Pull request merged**